azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/ILeaseManager.java

```diff
@@ -41,58 +42,53 @@
 	 * 
 	 * @return true if it does, false if not
+	 * @throws ExceptionWithAction with action EventProcessorHostActionStrings.CHECKING_LEASE_STORE on error
 	 */
-    public Future<Boolean> leaseStoreExists();
+    public CompletableFuture<Boolean> leaseStoreExists();
 
     /**
      * Create the lease store if it does not exist, do nothing if it does exist.
      * 
-     * @return true if the lease store already exists or was created successfully, false if not
+     * @return Void
+     * @throws ExceptionWithAction with action EventProcessorHostActionStrings.CREATING_LEASE_STORE on error
      */
-    public Future<Boolean> createLeaseStoreIfNotExists();
+    public CompletableFuture<Void> createLeaseStoreIfNotExists();
     
     /**
      * Not used by EventProcessorHost, but a convenient function to have for testing.
      * 
      * @return true if the lease store was deleted successfully, false if not
      */
-    public Future<Boolean> deleteLeaseStore();
-
-    /**
-     * Return the lease info for the specified partition. Can return null if no lease has been
-     * created in the store for the specified partition.
-     * 
-     * @param partitionId  id of partition to get lease for
-     * @return  lease info for the partition, or null
-     */
-    public Future<Lease> getLease(String partitionId);
+    public CompletableFuture<Boolean> deleteLeaseStore();
     
     /**
      * Return the lease info for all partitions.
      * 
      * A typical implementation could just call getLease() on all partitions.
      * 
-     * @return  Iterable list of lease info.
+     * @return  List of lease info.
      */
-    public Iterable<Future<Lease>> getAllLeases() throws Exception;
+    public CompletableFuture<List<Lease>> getAllLeases();
 
     /**
      * Create in the store the lease info for the given partition, if it does not exist. Do nothing if it does exist
      * in the store already.
      * 
      * @param partitionId  id of partition to create lease info for
-     * @return  the existing or newly-created lease info for the partition
+     * @return the existing or newly-created lease info for the partition
+     * @throws ExceptionWithAction with action EventProcessorHostActionString.CREATING_LEASE on error 
      */
-    public Future<Lease> createLeaseIfNotExists(String partitionId);
+    public CompletableFuture<Lease> createLeaseIfNotExists(String partitionId);
 
     /**
      * Delete the lease info for the given partition from the store. If there is no stored lease for the given partition,
      * that is treated as success.
      *  
      * @param lease  Lease info for the desired partition as previously obtained from getLease()
-     * @return  Void
+     * @return void
+     * @throws ExceptionWithAction with action EventProcessorHostActionString.DELETING_LEASE on error 
      */
-    public Future<Void> deleteLease(Lease lease);
+    public CompletableFuture<Void> deleteLease(Lease lease);
```

Comment:
>deleteLease [](start = 35, length = 11)

remove if it is not needed for eph. 
for testability consider creating separate interface implementing ILeaseManager...